### ISSUE TITLE
Update a11y-checker ruleset to 1July2023

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#14135]
On July 1 a new ruleset was published for accessibility-checker. It caused some failures so we reverted back to the older ruleset to unblock our biweekly release and all other PRs, #14134.

The ruleset needs to be updated and the resulting problems fixed. It appears that the bulk of the issues come from certain elements not having an aria-label on them. There's also two OverflowMenu tests that pass the same ID to the `toHaveNoACViolations()` function.

**Comment by lee-chase** [comment]
E.g. close button in composed modal

[Buttons should not be focusable when in subree of aria-](https://able.ibm.com/rules/archives/latest/doc/en-US/aria_hidden_nontabbable.html#%7B%22message%22%3A%22Element%20%5C%22button%5C%22%20should%20not%20be%20focusable%20within%20the%20subtree%20of%20an%20element%20with%20an%20'aria-hidden'%20attribute%20with%20value%20'true'%22%2C%22snippet%22%3A%22%3Cbutton%20type%3D%5C%22button%5C%22%20aria-label%3D%5C%22Close%5C%22%20title%3D%5C%22Close%5C%22%20class%3D%5C%22cds--modal-close%20c4p--tearsheet__header--no-close-icon%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_hidden_nontabbable%22%7D)

**Comment by tay1orjones** [comment]
> we might want to enforce that one of aria-label, aria-labelledby or title is supplied to the affected components as to enforce accessibility compliance


Here's my take on this - we've enforced accessibility via required props before (like in notification components), but in this case I think the complexity outweighs the benefits.
 * Validating conditionally that one of three props is provided is cumbersome via proptype validator functions and very difficult to achieve in TS as well.
* Updating props to be required is considered a breaking change. We could provide defaults for these, but imo a default generic label is a worse situation than not having a label at all.
* Ultimately, Carbon isn't the enforcer of accessibility standards, accessibility-checker is. accessibility-checker will prompt devs to put a label on these elements.

We can of course make sure to provide the props necessary to be compliant and provide valid examples in our storybook and tests. We also can document this requirement where possible so it's clear folks need to provide a label.

**Comment by lee-chase** [comment]
I mostly agree Tyler. There are cases such as the composed modal where a user might well end up duplicating a label because they can’t be labelled by something they have offloaded rendering to the ModalHeader for instance. I’d be interested in knowing if this duplication is frowned upon, or acceptable. That said I don’t have a solution in my head that I am happy with for exchanging Ids.

Otherwise Carbon should probably only be in the business of ensuring components can be rendered with a11y issues (not currently the case for Modal when hidden).
